modular_skyrat/master_files/code/modules/client/preferences/headshot.dm

```diff
@@ -0,0 +1,41 @@
+/datum/preference/text/headshot
+	category = PREFERENCE_CATEGORY_NON_CONTEXTUAL
+	savefile_identifier = PREFERENCE_CHARACTER
+	savefile_key = "headshot"
+	/// Assoc list of ckeys and their link, used to cut down on chat spam
+	var/list/stored_link = list()
+
+/datum/preference/text/headshot/apply_to_human(mob/living/carbon/human/target, value, datum/preferences/preferences)
+	target.dna.features["headshot"] = value
+
+/datum/preference/text/headshot/is_valid(value)
+	if(!length(value)) //Just to get blank ones out of the way
+		return TRUE
+	if(!findtext(value, "https://"))
+		to_chat(usr, span_warning("You need \"https://\" in the link!"))
+		return
+	if(!findtext(value, ".png") && !findtext(value, ".jpg"))
+		to_chat(usr, span_warning("You need either \".png\" or \".jpg\" in the link!"))
+		return
+	if(!findtext(value, "https://imgur.com", 1, 18) && !findtext(value, "https://i.gyazo.com", 1, 20) && !findtext(value, "https://media.discordapp.net", 1, 29))
```

Comment:
I wonder, wouldn't regex be more efficient here? I think it could be interesting to ask the big smarts at /tg/ or on coderbus about this, just for future reference.

Reply:
`Uploading Siezure.jpg.gif...`